modules/ding_provider/ding_provider.module

```diff
@@ -587,11 +590,16 @@ function ding_provider_get_pseudo_entity($ding_entity_id) {
   $ting_entity->ding_entity_id = $ding_entity_id;
   $ting_entity->provider_id = $local_id;
   $ting_entity->ding_entity_type = 'ding_entity';
-  $ting_entity->reply = new TingClientObject();
+  // TODO BBS-SAL: Clean out the direct property sets when done.
+  $ting_entity->reply = new TingPseudoObject();
   $ting_entity->reply->id = $ding_entity_id;
+  $ting_entity->reply->setDingId($ding_entity_id);
   $ting_entity->reply->localId = $local_id;
+  $ting_entity->reply->setLocalId($local_id);
```

Comment:
Provider id will also be a poor name now as the purpose of what we are doing is to implement a search provider. Local id is used based on an implicit understanding that the library system is a local system while OpenSearch spans multiple library organisations. This distinction also no longer holds with this abstration in place as we are no longer opensearch dependent.